ext/spl/php_spl.c

```diff
@@ -660,8 +660,8 @@ PHPAPI zend_string *php_spl_object_hash(zend_object *obj) /* {{{*/
 	intptr_t hash_handle, hash_handlers;
 
 	if (!SPL_G(hash_mask_init)) {
-		SPL_G(hash_mask_handle)   = (intptr_t)(php_mt_rand() >> 1);
-		SPL_G(hash_mask_handlers) = (intptr_t)(php_mt_rand() >> 1);
+		SPL_G(hash_mask_handle)   = 0;
+		SPL_G(hash_mask_handlers) = 0;
```

Comment:
You should remove these globals entirely, if they're going to be always zero.

Reply:
I will if tests pass. Why was it there at all?

Reply:
Historically, because this used to expose pointer values, and the randomness was used to obfuscate the exact pointer address. Nowadays, this no longer exposes pointers.

Reply:
Removed. I also changed the length to 16 bytes.

What about deprecating `spl_object_hash` in PHP 8.1?